tests/unit/h/services/annotation_write_test.py

```diff
@@ -232,6 +232,11 @@ def test_unhide(self, annotation, svc):
 
         assert not annotation.is_hidden
 
+    def test_upsert_annotation_slim_with_deleted_group(self, annotation, svc):
+        annotation.groupid = "deleted group"
+
+        svc.upsert_annotation_slim(annotation)
```

Comment:
I've seen advice saying that to assert "it doesn't raise" this should be enough.